segbot_arm_joystick/src/segbot_arm_joystick.cpp

```diff
@@ -0,0 +1,505 @@
+#include <ros/ros.h>
+#include <ros/package.h>
+
+//used for registering Ctrl-C events
+#include <signal.h>
+
+//ROS messages
+#include <sensor_msgs/JointState.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+
+//ROS action client used for calling actions
+#include <actionlib/client/simple_action_client.h>
+
+//JACO messages and actions
+#include <jaco_msgs/FingerPosition.h>
+#include <jaco_msgs/HomeArm.h>
+
+//our own arm library 
+#include <segbot_arm_manipulation/arm_utils.h>
+
+#include <sensor_msgs/Joy.h>
+#include <bwi_services/SpeakMessage.h>
```

Comment:
It's OK for segbot_arm to depend on a package in bwi_common. Just do the dependency in the normal way.